src/ai/ai.cpp

```diff
@@ -81,69 +107,66 @@ void form_alliances(sys::state& state) {
+				if(other.get_is_player_controlled() == false && !(other.get_overlord_as_subject().get_ruler())  && !nations::are_allied(state, n, other) && !military::are_at_war(state, other, n) && ai_will_accept_alliance(state, other, n))
 					alliance_targets.push_back(other.id);
-				}
 			}
 
 			if(!alliance_targets.empty()) {
 				std::sort(alliance_targets.begin(), alliance_targets.end(), [&](dcon::nation_id a, dcon::nation_id b) {
-					if(state.world.nation_get_military_score(a) != state.world.nation_get_military_score(b))
-						return state.world.nation_get_military_score(a) > state.world.nation_get_military_score(b);
+					if(estimate_strength(state, a) != estimate_strength(state, b))
+						return estimate_strength(state, a) > estimate_strength(state, b);
 					else
 						return a.index() > b.index();
 				});
-
 				nations::make_alliance(state, n, alliance_targets[0]);
 			}
 		}
 	}
 }
 
+static inline bool ai_is_close_enough(sys::state& state, dcon::nation_id target, dcon::nation_id from) {
+	auto target_continent = state.world.province_get_continent(state.world.nation_get_capital(target));
+	auto source_continent = state.world.province_get_continent(state.world.nation_get_capital(from));
+	return (target_continent == source_continent) || bool(state.world.get_nation_adjacency_by_nation_adjacency_pair(target, from));
+}
+
 bool ai_will_accept_alliance(sys::state& state, dcon::nation_id target, dcon::nation_id from) {
 	if(!state.world.nation_get_ai_is_threatened(target))
 		return false;
 
 	if(state.world.nation_get_ai_rival(target) == from || state.world.nation_get_ai_rival(from) == target)
 		return false;
-
-	auto target_continent = state.world.province_get_continent(state.world.nation_get_capital(target));
-	auto source_continent = state.world.province_get_continent(state.world.nation_get_capital(from));
-
-	bool close_enough = (target_continent == source_continent) || bool(state.world.get_nation_adjacency_by_nation_adjacency_pair(target, from));
-
-	if(!close_enough)
+	
+	// Same rival equates to instantaneous alliance (we benefit from more allies against a common enemy)
+	if(state.world.nation_get_ai_rival(target) == state.world.nation_get_ai_rival(from))
+		return true;
+	
+	// Otherwise we may consider alliances only iff they are close to our continent or we are adjacent
+	if(!ai_is_close_enough(state, target, from))
 		return false;
-
-	auto target_score = state.world.nation_get_military_score(target);
-	auto source_score = state.world.nation_get_military_score(from);
-
-	return source_score * 2 >= target_score;
+	
+	// And also if they're powerful enough to be considered for an alliance
+	auto target_score = estimate_strength(state, target);
+	auto source_score = estimate_strength(state, from);
+	return source_score * 2.5f >= target_score;
```

Comment:
Is there a reason you have increased the range here? Since we can't see how well the AI is performing yet, please don't fiddle with numbers like this when we can't see whether the change will be better or worse. Otherwise, the change is just a change for the sake of change